File: app/src/main/java/com/globalwarming/earthsaver/LoginActivity.java
package com.globalwarming.earthsaver;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.util.Patterns;
import android.widget.Button;
import android.widget.TextView;
import com.google.android.material.snackbar.Snackbar;
import com.google.android.material.textfield.TextInputLayout;
import com.google.firebase.auth.FirebaseAuth;

public class LoginActivity extends AppCompatActivity {

    private Toolbar toolbar;
    private TextInputLayout etEmail;
    private TextInputLayout etPassword;
    private Button buttonLogin;
    private TextView textRegister;
    private TextView textForgotPassword;

    private FirebaseAuth mAuth;

    private ProgressDialog progressDialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        toolbar = findViewById(R.id.toolbar);
        etEmail = findViewById(R.id.etEmail);
        etPassword = findViewById(R.id.etPassword);
        buttonLogin = findViewById(R.id.buttonLogin);
        textRegister = findViewById(R.id.textRegister);
        textForgotPassword = findViewById(R.id.textForgotPassword);
        setSupportActionBar(toolbar);

        progressDialog = new ProgressDialog(this);
        progressDialog.setMessage("Logging in ...");
        progressDialog.setCanceledOnTouchOutside(false);
        progressDialog.setCancelable(false);

        mAuth = FirebaseAuth.getInstance();

        textRegister.setOnClickListener(v -> {
            Intent intent = new Intent(LoginActivity.this, RegistrationActivity.class);
            startActivity(intent);
        });

        textForgotPassword.setOnClickListener(v -> {
            Intent intent = new Intent(this, ForgotPasswordActivity.class);
            startActivity(intent);
        });

        buttonLogin.setOnClickListener(v -> {
            Util.hideKeyboard(v);
            if (validate()) {
                String email = etEmail.getEditText().getText().toString().trim();
                String password = etPassword.getEditText().getText().toString().trim();
                progressDialog.show();

                mAuth.signInWithEmailAndPassword(email, password).addOnCompleteListener(task -> {
                    progressDialog.dismiss();
                    if (task.isSuccessful()) {
                        //Login is complete
                        Intent intent = new Intent(LoginActivity.this, HomeActivity.class);
                        startActivity(intent);
                        finish();
                    } else {
                        //Login is not complete
                        Snackbar.make(etEmail, task.getException().getMessage(), Snackbar.LENGTH_LONG).show();
                    }
                });
            }
        });

    }

    private Boolean validate() {
        String email = etEmail.getEditText().getText().toString().trim();
        String password = etPassword.getEditText().getText().toString().trim();

        //Email
        if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            Snackbar.make(etEmail, "Email is invalid.", Snackbar.LENGTH_SHORT).show();
            return false;
        }
        //Password
        if (password.length() < 6) {
            Snackbar.make(etPassword, "Password has to e of 6 characters", Snackbar.LENGTH_SHORT).show();
            return false;
        }
        return true;
    }

    @Override
    protected void onDestroy() {
        if (progressDialog.isShowing()) {
            progressDialog.dismiss();
        }
        super.onDestroy();
    }

}

File: app/src/main/java/com/globalwarming/earthsaver/LoginActivity.kt
package com.globalwarming.earthsaver

import android.content.Intent
import android.os.Bundle
import android.util.Patterns
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.globalwarming.earthsaver.databinding.ActivityLoginBinding
import com.globalwarming.earthsaver.util.Loader
import com.google.android.gms.tasks.Task
import com.google.android.material.snackbar.Snackbar
import com.google.firebase.auth.AuthResult
import com.google.firebase.auth.FirebaseAuth

class LoginActivity : AppCompatActivity() {

    private lateinit var mAuth: FirebaseAuth
    private lateinit var loader: Loader
    private lateinit var binding: ActivityLoginBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_login)
        setSupportActionBar(binding.toolbar)

        loader = Loader(this, "Logging in ...")
        mAuth = FirebaseAuth.getInstance()

        binding.textRegister.setOnClickListener {
            val intent = Intent(this@LoginActivity, RegistrationActivity::class.java)
            startActivity(intent)
        }
        binding.textForgotPassword.setOnClickListener {
            val intent = Intent(this, ForgotPasswordActivity::class.java)
            startActivity(intent)
        }
        binding.buttonLogin.setOnClickListener {
            Util.hideKeyboard(it)
            if (validate()) {
                val email = binding.etEmail.editText!!.text.toString().trim()
                val password = binding.etPassword.editText!!.text.toString().trim()
                loader.show()
                mAuth.signInWithEmailAndPassword(email, password)
                    .addOnCompleteListener { task: Task<AuthResult?>? ->
                        loader.dismiss()
                        if (task?.isSuccessful == true) {
                            //Login is complete
                            val intent = Intent(this@LoginActivity, HomeActivity::class.java)
                            startActivity(intent)
                            finish()
                        } else {
                            //Login is not complete
                            Snackbar.make(
                                binding.etEmail,
                                task?.exception?.message.toString(),
                                Snackbar.LENGTH_LONG
                            ).show()
                        }
                    }
            }
        }
    }

    private fun validate(): Boolean {
        val email = binding.etEmail.editText!!.text.toString().trim()
        val password = binding.etPassword.editText!!.text.toString().trim()

        //Email
        if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            Snackbar.make(binding.etEmail, "Email is invalid.", Snackbar.LENGTH_SHORT).show()
            return false
        }
        //Password
        if (password.length < 6) {
            Snackbar.make(binding.etPassword, "Password has to e of 6 characters", Snackbar.LENGTH_SHORT).show()
            return false
        }
        return true
    }

    override fun onDestroy() {
        if (loader.isShowing) {
            loader.dismiss()
        }
        super.onDestroy()
    }
}